### ISSUE TITLE
Use phone keyboard in emulator

### ISSUE THREAD

**Issue by PierreB105** [numworks/epsilon#1879]
Typing python code is hard on the phone, could you add the possibility to use the regular phone keyboard to input text? Thanks

**Comment by mobluse** [comment]
I don't understand what you mean by "regular phone keyboard". Do you have a phone with physical buttons i.e. not only a touch screen? Or do you mean a Bluetooth keyboard or USB-OTG keyboards? I think support for Bluetooth keyboard would be an advantage, also I don't mind support for physical phone keyboards or USB-OTG keyboards for the phone apps.

**Comment by PierreB105** [comment]
My idea is to be able to use the phone built in keyboard (the one I'm using now to answer your question) rather than having to use NumWorks keys. This allows to avoid having to use ALPHA or shift to input letters or parenthesis for instance. 
This may not be so useful for calculation, but much faster when typing Python code. 
Unless this is already possible but didn't find how to do!

**Comment by RedGl0w** [comment]
It seems to be a duplicate of https://github.com/numworks/epsilon/issues/1274

**Comment by mobluse** [comment]
I thought you meant the apps for Android and iPhone, but I now understand you mean the NumWorks site in a browser running on a phone. Most things there work with a Bluetooth keyboard. The only thing that didn't work with a BT keyboard is when you edit a script in *My scripts* and want to type in the display. Then even if you try to select the display it doesn't get selected when it's next to the editor. I have my iPhone in horizontal position and use Chrome as browser. If you then Save you can type using BT-keyboard in the single display without the editor. 

Though if you don't have a BT keyboard you cannot type in the displays using the pop up phone keyboard, and it's difficult to edit because each time you type the window is homed, and you can't see what you type.

**Comment by PierreB105** [comment]
Hi, I was talking about the iPhone app. If there is a way to use the regular phone keyboard, I didn't find it.

**Comment by mobluse** [comment]
I see, and the iPhone app also doesn't support Bluetooth keyboard.

**Comment by PierreB105** [comment]
So just to be sure (sorry about the quiproquo, I didn't understand you were thinking about the Bluetooth keyboard), there is currently no way of using the NumWorks app as on the picture I made in the previous message?

**Comment by mobluse** [comment]
No, not the current iPhone app, but you can edit a script (My scripts) in the browser of the phone (tested in Chrome) and run it without a BT keyboard, but it is difficult on my phone because when you enter a character the browser scrolls to the top, but maybe it works better on an iPad.
